platform/ios/src/MGLMapView.mm

```diff
@@ -1711,6 +1719,12 @@ - (void)handleRotateGesture:(UIRotationGestureRecognizer *)rotate
 
     [self cancelTransitions];
 
+    if (MGLDegreesFromRadians(self.currentRotation) < self.rotationThreshold && self.isZooming) {
+        rotate.delaysTouchesBegan = YES;
+        self.currentRotation += abs(rotate.rotation);
+        rotate.rotation = 0;
+    }
```

Comment:
I notice that this doesn’t apparently do anything to return early. Should the rest of `handleRotateGesture:` still execute if the actual rotation is being blocked?

Reply:
I chose not to add a return to that check due to a couple of small issues: 
1. It triggers an `NSException` when the rotate gesture ends.
2. It can make the map jump to wherever the touches are currently occurring. 

Continuing the gesture with a `rotation` value of `0` makes the rotation smoother once the rotation threshold is met, and prevents the exception.

Reply:
I think those are issues we should fix rather than ignore, given that the alternative is to pretend that a rotation gesture is happening and do all of the (unnecessary) work — like canceling transitions, changing the user tracking mode, sending camera change notifications, and so on.